modal-go/sandbox.go

```diff
@@ -214,6 +220,44 @@ func (sb *Sandbox) Tunnels(timeout time.Duration) (map[int]*Tunnel, error) {
 	return sb.tunnels, nil
 }
 
+// Poll checks if the Sandbox has finished running.
+// Returns nil if the Sandbox is still running, else returns the exit code.
+func (sb *Sandbox) Poll() (*int32, error) {
+	resp, err := client.SandboxWait(sb.ctx, pb.SandboxWaitRequest_builder{
+		SandboxId: sb.SandboxId,
+		Timeout:   0,
+	}.Build())
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.GetResult() != nil && resp.GetResult().GetStatus() != pb.GenericResult_GENERIC_STATUS_UNSPECIFIED {
+		sb.result = resp.GetResult()
+	}
+
+	return sb.ReturnCode(), nil
+}
+
+// ReturnCode returns the exit code of the Sandbox process if it has finished running, else nil.
```

Comment:
This API is a bit surprising to me as well since you could call `sb.ReturnCode()` and it wouldn't actually tell you if the sandbox has exited or not, it would only return something if Wait() or Poll() has previously been called.